src/Compilers/VisualBasic/Test/Syntax/TestSyntaxNodes.vb

```diff
@@ -145,6 +145,22 @@ Namespace Microsoft.CodeAnalysis.VisualBasic.UnitTests
             Assert.False(xc <> children, "Verifying <> operator for ChildSyntaxList items - This should return false as xc was assigned from Children")
         End Sub
 
+        <Fact>
+        <WorkItem(21812, "https://github.com/dotnet/roslyn/issues/21812")>
+        Public Sub TestTupleTypeInSyntaxFactory()
```

Comment:
>TestTupleTypeInSyntaxFactory [](start = 19, length = 28)

I think it would be good to pair factory API test with an attempt to parse similar construct as text and verify consistent behavior between the factory and the parser (by checking if parser creates tuple type node) for each target scenario. Should probably also cover Object creation, Inherits and implements.